src/components/searchForm/searchForm.js

```diff
@@ -1,58 +1,117 @@
 import React from 'react';
 import queryString from 'query-string';
 import './searchForm.css'
-import RecipesList from "../recipesList/recipesList";
-import Varenyky from "../../img/varenyky.jpg";
+import RecipesList from "../recipesList";
+import {storeRecipes, storeSearchResults} from "../../actions/creators";
+import {connect} from "react-redux";
 
 
-export default class SearchForm extends React.Component {
+class SearchForm extends React.Component {
     constructor(props) {
         super(props);
+        this.buttonClicked = true;
+        this.updateCounter = 0; // Shit, but easy
         this.state = {
-            values: queryString.parse(props.location.search)
+            ...{
+                searchResults: [],
+                sortBy: "name",
+                maxTime: Infinity,
+                query: ''
+            },
+            ...queryString.parse(props.location.search)
         };
     }
 
+    handleFormSubmit(ev) {
+        ev.preventDefault();
+        this.buttonClicked = true;
+        this.forceUpdate();
+    }
 
-    render() {
-        // Just temporary filling
-        const recipes = Array.from({length: 20}, (e, i) => ({
-            description: 'Varenyky!!! One of the best dishes, created by people',
-            photoSource: Varenyky,
-            title: 'Varenyky',
-            likes: i * 5794 % 157,
-            id: i
-        }));
+    handleQuery(ev) {
+        this.setState({"query": ev.target.value});
+    }
+
+    handleOrder(ev) {
+        this.setState({"sortBy": ev.target.value});
+    }
 
+    handleCookingTime(ev) {
+        this.setState({"maxTime": ev.target.value});
+    }
+
+    shouldComponentUpdate(nextProps, nextState, nextContext) {
+        return this.buttonClicked && this.updateCounter <= 2
+    }
 
+    getRecipes() {
+        // Here it would be better to do this on API side rather than fetching everything and filtering.
+        // But let it be like this for now
+
+        fetch(process.env.REACT_APP_DB_SERVER + "/recipes").then(response => response.json()).then(data => {
```

Comment:
You shouldn't be getting all recipes. json-server supports querying, please try this.